make_project_table.py

```diff
@@ -0,0 +1,123 @@
+
+    with open(".personal-github-token") as fl:
+        GH_TOKEN = fl.read()
+
+    g = Github(GH_TOKEN)
+    repo = g.get_repo("HERA-Team/hera-validation")
+
+    projects = repo.get_projects()
+
+    # issues = repo.get_issues(state='all')
+    # prs = repo.get_pulls(state='all')
+    #
+    # # Isolate issues/prs that define formal tests/steps
+    # steps = [issue for issue in everything if "formal-test" in [lbl.name for lbl in issue.labels]]
+    #
+    # everything = [issue for issue in issues] + [pr for pr in prs]
+    # This finds anything with the pattern X.X.X in a string (i.e. the STEP.MAJOR part of a
+    # step identifier)
+    step_number_pattern = re.compile(r"(-?\d+(\.\d+){0,2})")
+
+    tables = """## Project Plan
+Here follows a formal plan for envisaged tests, automatically generated from our Projects. 
+This list is *not final*: it may be extended or 
+modified at any time. However, it provides a reasonable snapshot of the current status and plans of 
+the validation team.
+
+What the symbols mean:
+
+Sym. | Meaning | Sym. | Meaning | Sym.  |Meaning
+-------| ----- | ----- | ---- | ----- | ------
+:egg:      | Idea in gestation  | :hammer:   | Currently being worked on | :thinking: | PR submitted... 
+:heavy_check_mark: | Passed     | :x:        | Failed | :watch:    | Not required for H1C IDR2
+
+"""
+
+    for proj in sorted(projects, key=lambda p: p.name):
+
+        cols = proj.get_columns()
+
+        steps = sum([list(col.get_cards()) for col in cols], [])
+        steps = [step.get_content() for step in steps]
+
+        step_dict = {}
+
+        for step in steps:
+            if step.__class__.__name__ not in ["Issue", "PullRequest"]:
+                print("card ", step, " has class", step.__class__.__name__)
+                continue
+            # get status
+            if step.milestone is None or step.milestone.title != "H1C IDR2":
+                status = ":egg:"
+                status_num = 0
+            elif step.__class__.__name__ == "Issue" and step.state == 'open':
+                status = ":hammer:"
+                status_num = 1
+            elif step.__class__.__name__ == "PullRequest" and step.state == "open":
+                status = ":thinking:"
+                status_num = 2
+            elif step.state == "closed":
+                status = ":heavy_check_mark:"
+                status_num = 3
+            else:
+                raise ValueError("status not recognized for ", step)
+
+            # Get step number
+            step_number = step_number_pattern.search(step.title).group()
+            major_step = ".".join(step_number.split(".")[:2])  # only up to X.X
+            description = step.title.split(step_number)[-1].split(":")[-1].split('--')[-1]
+            if description:
+                description = ": " + description
+
+            # Get title
+            title = f"[{major_step}]({step.url})"
```

Comment:
This particular line will only execute in python >=3.4. That won't be a _huge_ issue if you only expect users to have up to date python but if so it is worth stating.

Reply:
I'll just make it explicit in the README/script that you need py34+. I think everyone is using that now... and this script is not vital, just useful.